### PR TITLE
Add pre-commit autoupdate workflow

### PR DESCRIPTION
**Pull request by mikealfare**
### Problem

We moved all linting and formatting dependencies into `.pre-commit-config.yml`. While this consolidates these tools and isolates the dev environment from these dependencies, it also hides them from `dependabot`.

### Solution

Add a reusable workflow in `dbt-labs/actions` and call it here. This workflow will:
- run `pre-commit autoupdate`
- run `pre-commit run --all-files`
- submit a PR if updates were found

CI will take care of the verification of whether new conflicts were found in the updated `pre-commit` hooks.

The resulting PR: https://github.com/dbt-labs/dbt-adapters/pull/297

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-adapters/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have run this code in development, and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc.) or this PR has already received feedback and approval from Product or DX

### CODE REVIEW

File: .github/workflows/precommit-autoupdate.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  contents: write
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name }}-${{ github.sha }}
+  cancel-in-progress: true
+
+jobs:
+  precommit-autoupdate:
+    name: "Run pre-commit autoupdate"
+    uses: dbt-labs/actions/.github/workflows/pre-commit-autoupdate.yml@pre-commit-auto-update
```
Comment (emmyoop) [inline .github/workflows/precommit-autoupdate.yml:18]: Remember to update this to `@main`

**Review by emmyoop — APPROVED** [review]
Looks good pending branch update

File: .github/workflows/precommit-autoupdate.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    secrets:
+      TOKEN: ${{ secrets.FISHTOWN_BOT_PAT }}
+      SLACK_WEBHOOK_URL: ${{ secrets.SLACK_DEV_ADAPTER_PULL_REQUESTS }}
```
Comment (mikealfare) [inline .github/workflows/precommit-autoupdate.yml:21]: This needs to be added to the repo secrets.

**Review by VersusFacit — APPROVED** [review]
Thanks for adding a test build.

**Comment by mikealfare** [comment]
I need to wait until the new secrets are added and then this can be merged.

**Pull request merged**